rework metrics again (metadata.yaml doesn't match)

zenmoto opened the pull request:
Draft PR to allow discussion on metric schema

zenmoto committed:
client wrapper for MongoDB Atlas support

Adds pagination, error handling, will be extended to manage metric
transformation.

zenmoto committed:
rework metrics again (metadata.yaml doesn't match)

rmfitzpatrick commented on receiver/mongodbatlasreceiver/internal/metadata/generated_metrics.go:
```diff
@@ -0,0 +1,580 @@
+}
+
+// Intentionally not exposing this so that it is opaque and can change freely.
+type metricImpl struct {
```
Not a comment on the substance of metrics produced, but I wonder if it would be easier to manage a slice of structs with name, description, unit, and datatype fields that's used by a metric producer.  If necessary the struct could have an initializer function field that's used for unique items.  If `MetricIntf` has a larger intended usage then maybe the reduced redundancy isn't worth it though.

zenmoto replied:
Yeah, I keep getting the urge to do that, but since many of these metricImpl instances match to a single pdata.Metric instance, I'd end up duplicating the data. I might be able to build a different struct that did a better job of keeping it all together though.

rmfitzpatrick commented on receiver/mongodbatlasreceiver/internal/metadata/metric_name_mapping.go:
```diff
@@ -0,0 +1,1001 @@
+func inferMetricType(mongoType string) (pdata.MetricDataType, string) {
+	var dataType pdata.MetricDataType
+	switch mongoType {
+	case "BYTES", "MEGABYTES", "SECONDS", "KILOBYTES":
```
Could these also be used to set the unit on the metric?

zenmoto replied:
They could. My initial intent was to use these values as a fallback for any not-currently-documented metric that gets added to Atlas from here on out, but I'm thinking that may be going too far and I should cut this functionality. I think that the metric format from Atlas differs enough from OTel standards that it might be counter-productive to pass them through without some more knowledge about them. I've also gotten some advice to keep this simple, and that functionality seems to be something that would be better added if needed later.

rmfitzpatrick commented on receiver/mongodbatlasreceiver/internal/metadata/metric_name_mapping.go:
```diff
@@ -0,0 +1,1001 @@
+
+	// sfx: process.cpu.user
+	"PROCESS_CPU_USER": {"process.cpu.utilization", map[string]pdata.AttributeValue{
+		"state":       pdata.NewAttributeValueString("user"),
```
I'm not familiar with Atlas, but are there distinguishing values in the measurement that can be used?  Having a ~callback for the measurement and metric similar to the initializer function for settings attributes may be helpful if so.

zenmoto replied:
Unfortunately much of the attribute definitions here come from the name of the measurement, and the names aren't quite consistent enough to be parsable for the information. The measurement data structure does have the unit name, but there's nothing I can really use to fold multiple measurements into a single metric.

zenmoto committed:
align metrics in metadata.yaml with name map

zenmoto committed:
start using multierror

zenmoto committed:
poll should return an error, not log

zenmoto committed:
changes to work with scraperhelper

mx-psi committed:
[exporter/datadogexporter] Remove spammy logging (#5856)

KSerrania committed:
[datadogexporter] Remove obsolete report_buckets config (#5858)

* [datadogexporter] Remove obsolete report_buckets config

* Add comments to config unmarshall

bogdandrutu committed:
Promote Alex Boten to maintainer (#5865)

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

vikrambe committed:
Add support for composite sampling policy to the tailsampler (#4958)

Is your feature request related to a problem? Please describe.
#1410
Design Doc - Added support for composite policy in tailsampling processor. This would help in grouping sampling policies and rate limiting them. https://docs.google.com/document/d/10wpIv3TtXgOik05smHm3nYeBX48Bj76TCMxPy8e1NZw/edit#heading=h.ecy5l2puwtp4
This is a split. Refer PR open-telemetry/opentelemetry-collector#1894 (comment)

Due to EasyCLA issue opening a new PR: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/4396

Link to tracking Issue: 1306

tigrannajaryan committed:
Improve performance of metric expression matcher (#5864)

Refactored env struct to avoid creating closure functions for every
expression evaluation operation. This reduced the allocations and
increased the speed.

Improvements:

$ benchstat old.txt new.txt
name          old time/op    new time/op    delta
ExprFilter-8     537µs ± 2%     476µs ± 1%  -11.33%  (p=0.000 n=9+10)

name          old alloc/op   new alloc/op   delta
ExprFilter-8     151kB ± 0%     102kB ± 0%  -32.43%  (p=0.000 n=8+8)

name          old allocs/op  new allocs/op  delta
ExprFilter-8     8.44k ± 0%     6.39k ± 0%  -24.24%  (p=0.000 n=10+10)

zenmoto committed:
fix error handling with scraper, align metrics

bogdandrutu committed:
[deltatocomulativeprocessor]: Move tracking to internal (#5871)

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

bogdandrutu committed:
[cumulativetodeltaprocessor]: small nit, avoid creating object until is necessary (#5872)

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

Commit:
Update Redis receiver's metric names (#5837)

* Update Names & Tests

* Update db metric names

* Revert Change

Commit:
Migrated Google Cloud Spanner receiver to scraper approach. (#5868)

KSerrania committed:
[datadogexporter] Fix cumulative histogram handling in distributions mode (#5867)

* [datadogexporter] Fix cumulative histogram handling in distributions mode

* Remove unused testSketch

* Cosmetic fixes

sethAmazon committed:
Add Resource Poller Before TestBed Runs (#5671) (#5845)

zenmoto committed:
Add client wrapper for MongoDB Atlas support (#5386)

* client wrapper for MongoDB Atlas support

Adds pagination, error handling, will be extended to manage metric
transformation.

* address review comments

MongoDB Atlas wrapper returns errors rather than logging
change resource attributes to use conventions
modify error handling

* fix capitalization on error messages

* go mod tidy

* poll should return an error, not log

* changes to work with scraperhelper

Commit:
update dependencies per dependabot (#5907)

zenmoto committed:
remove sfx annotation comments

zenmoto committed:
break up poll function
